README.md

```diff
@@ -1,138 +1,615 @@
+
+```mjs
+// bound.mjs
+export decorator @bound {
+  @register((target, name) => {
+    const method = target[name];
+    Object.defineProperty(target, name, {
+      get() {
+        const bound = method.bind(this);
+        Object.defineProperty(this, name, { value: bound, configurable: true });
+      }
+      configurable: true
+    });
+  })
+}
+```
+
+There are various appraoches to writing an auto-bound decorator, but ultimately, the most efficient way may be built into the JavaScript engine; see [NEXTBUILTINS.md](./NEXTBUILTINS.md) for discussion of a built-in `@bound` decorator, and the [bound-decorator](https://github.com/mbrowne/bound-decorator) repository for another approach.
+
+### `@callable`
+
+The `@callable` decorator makes it possible to invoke the class without `new`. When a class decorated with `@callable` is called, its static `call` method is invoked.
+
+```mjs
+import { @callable } from "./callable.mjs";
+
+@callable
+class MyDate {
+  static call(...args) { return Date(...args) }
+  constructor(...args) { return new Date(...args) }
+}
+```
+
+An implementation in terms of `@wrap`:
+
+```mjs
+// callable.mjs
+
+decorator @call(callback) {
+  @wrap(klass => {
+    function subclass(...args) {
+      if (new.target === undefined) {
+        return callback.call(klass, ...args);
+      } else {
+        return Reflect.construct(klass, args, new.target);
+      }
+    }
+    subclass.__proto__ = klass;
+    subclass.prototype.__proto__ = klass;
+    return subclass;
+  })
+}
+
+export decorator @callable {
+  @call(function(...args) { return this.call(...args); })
+}
+```
+
+Note that a decorator like `@call` could be considered for a future built-in decorator, in a way that avoids creating an additional subclass.
+
+### Combined example
+
+Some of the above examples could be combined to form a mini-framework, to make it easier to write HTML Custom Elements.
+
+```mjs
+import { @set } from "./set.mjs";
+import { @tracked } from "./tracked.mjs";
+import { @bound } from "./bound.mjs";
+import { @defineElement } from "./defineElement.mjs";
+
+@defineElement('counter-widget')
+class CounterWidget extends HTMLElement {
+  @tracked x = 0;
+
+  @set onclick = this.clicked;
+
+  @bound clicked() { this.x++; }
+
   connectedCallback() { this.render(); }
 
-  render() {
-    this.textContent = this.x.toString();
-  }
+  render() { this.textContent = this.x.toString(); }
+}
+```
+
+## The idea
+
+A new goal of this proposal: It should be possible for tools and JS engines to understand what's going on with decorators.
+
+Core elements:
+- Decorators have `@` as part of their name; `@decorator` names form a separate namespace.
+- There's a set of built-in decorators that serve as the basic building blocks.
+- Developers to create their own decorators by composing other decorators.
+- Decorators cannot be treated as JavaScript values; they may only be applied in classes, composed, exported, imported, etc.
```

Comment:
why only classes? Since they're syntactic, they'd work on function declarations and expressions just fine, I think?

Reply:
Right, that's going to be proposed in NEXTBUILTINS.md.